src/Lib.hs

```diff
@@ -0,0 +1,72 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module Lib
+    (
+     ascii_art_str,
+     basic_str,
+     basic_const_zero,
+     BasicExternal,
+    ) where
+
+import Foreign.C.Types
+import Foreign.Ptr
+import Foreign.C.String
+import Foreign.Storable
+import Foreign.Marshal.Array
+import Foreign.ForeignPtr
+import Control.Applicative
+
+
+data BasicStruct = BasicStruct {
+    data_ptr :: Ptr ()
+}
+
+instance Storable BasicStruct where
+    alignment _ = 8
+    sizeOf _ = sizeOf nullPtr
+    peek basic_ptr = BasicStruct <$> peekByteOff basic_ptr 0
+    poke basic_ptr BasicStruct{..} = pokeByteOff basic_ptr 0 data_ptr
+
+
+type BasicExternal = ForeignPtr BasicStruct
+-- !an array of size 1 of type `BasicStruct`
+type BasicInternal = Ptr BasicStruct
+-- exported functions
+
+
+basic_const_zero :: IO BasicExternal
+basic_const_zero = do
+    basic <- create_basic
+    withForeignPtr basic basic_const_zero_raw
+    return basic
+
+basic_str :: BasicExternal -> IO String
+basic_str basic_external = withForeignPtr basic_external (\p -> basic_str_raw p >>= peekCString)
+
+-- |The `ascii_art_str` function prints SymEngine in ASCII art.
+-- this is useful as a sanity check
+ascii_art_str :: IO String
+ascii_art_str = ascii_art_str_raw >>= peekCString
+
+-- Unexported raw functions------------------------
+
+-- |Create a basic object that represents all other objects through
+-- the FFI
+create_basic :: IO BasicExternal
+create_basic = do
+    basic_ptr <- newArray [BasicStruct { data_ptr = nullPtr }]
+    basic_new_heap_raw basic_ptr
+    finalized_ptr <- newForeignPtr ptr_basic_free_heap_raw basic_ptr
+    return finalized_ptr
+
+
+
+foreign import ccall "cwrapper.h ascii_art_str" ascii_art_str_raw :: IO CString
```

Comment:
Because if SymEngine is installed to a standard path where the compiler searches for, then the user doesn't have to give the additional include dirs. For example if the user installs into /usr/include. If we use just cwrapper.h then the user has to give the additional include directory /usr/include/symengine.